woocommerce/woocommerce
# Replace hardcoded notices with the correct wp_print_notice function

### All Submissions:


The contribution guidelines do not cover changelog entry creation—I had an error from a pre-commit hook when pushing.





### Changes proposed in this Pull Request:

When working on https://github.com/woocommerce/woocommerce-blocks/pull/8659 we realised several of the notices rendered by WooCommerce core across several frontend template files did not use the notice function helpers (`wp_print_notice`) or templates, and were instead hardcoded.

This PR replaces hardcoded notices with the correct functions and makes it possible to modify the notices via template overrides.

Additionally, a small modification to `wp_print_notice` was added so that the notice could be returned as a string instead of rendered. This was to make it more flexible. e.g.

```php
$notice_html = wp_print_notice( 'Example', 'error', [], true );
```



### How to test the changes in this Pull Request:




These instructions cover how to test notices that were changed.

1. As a logged in customer with no orders, go to the My Account page
     - Click "orders" navigation item. Confirm the "no orders" notice is shown
     - Click "downloads" navigation item. Confirm the "no downloads" notice is shown
     - Navigate direct to "my-account/payment-methods". Confirm the "no methods" notice is shown
     - Navigate direct to "my-account/add-payment-method". Confirm the notice is shown.
2. Go to the cart page (ensure there are no items in your cart). Confirm the "empty cart" notice is shown.
3. Add an item to your cart.
4. Go back to the cart page. 
5. Remove the item from your cart. Confirm you see both an "undo" notice, and an "empty cart" notice.
6. Go to the shop page. Append `?s=noresults` to the URL. Confirm you see the "No products were found matching your selection" notice.
7. Add some items to your cart then place an order via the checkout
7. Go back to the My Account Page
8. Click the "orders" navigation item
9. Click the button to view an order
10. Change the URL and replace the order ID there with something invalid e.g. 123
11. Confirm you see the invalid order notice



### Other information:




### FOR PR REVIEWER ONLY:
